src/template_utils.py

```diff
@@ -0,0 +1,56 @@
+import os
+import re
+from typing import Any, Dict
+
+import yaml
+
+
+def load_yaml_template(template_path: str) -> str:
```

Comment:
Do you need the 2 extra sub-methods here? They're not called anywhere AFAICT

Reply:
yup. updated those to private methods.